book/image-controller.md

```diff
@@ -1,2 +1,141 @@
 # Image Controller
 
+## Overview
+
+The Image Controller for AppStudio helps set up container image repositories
+for AppStudio Components. With image controller, AppStudio ensures one image
+repository per Component and each build pipeline will have dedicated secret to
+push image. AppStudio provides an isolated environment for users to push and
+manage images.
+
+Image controller watches Components to take actions according to the deletion
+event and the value set to specific annotations. External services are able to
+interact with image controller via annotations:
+
+- `image.redhat.com/generate`: if set, image controller starts to set up image
+  repository on Quay.io for a Component, or attempts to switch repository
+  visibility if the repository exists already.
+
+- `image.redhat.com/image`: information of created resources is set to this
+  annotation as a JSON string. Error message will also be set to here if there
+  is.
+
+## Set up image repository
+
+To request the controller to setup an image repository, annotate the Component
+CR with `image.redhat.com/generate: '{"visibility": "public"}'` or
+`image.redhat.com/generate: '{"visibility": "private"}'` depending on desired
+repository visibility.
+
+Here is an example that a component is annotated for creating a public image
+repository:
+
+```yaml
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: Component
+metadata:
+  annotations:
+    image.redhat.com/generate: '{"visibility": "public"}'
+  name: billing
+  namespace: image-controller-system
+spec:
+  application: city-transit
+  componentName: billing
+```
```

Comment:
I would add line with three dots `...` to show that the CR definition is not complete.